packages/app/src/cli/services/deploy/upload.ts

```diff
@@ -240,6 +237,27 @@ ${output.token.json(userErrors)}
   return res.data.functionSet.function?.id as string
 }
 
+async function uploadWasmBlob(extension: FunctionExtension, apiKey: string, token: string): Promise<string> {
+  const {url, headers, maxSize} = await getFunctionExtensionUploadURL({apiKey, token})
+  headers['Content-Type'] = 'application/wasm'
+
+  const functionContent = fs.readFileSync(extension.buildWasmPath())
```

Comment:
I'd import `file` from `@shopify/cli-kit` and use the promise version of this API: `file.readFile`